pkg/prometheus/operator.go

```
@@ -2575,26 +2577,43 @@ func validateRemoteWriteSpec(spec monitoringv1.RemoteWriteSpec) error {
 	return nil
 }
 
-func validateRelabelConfig(rc monitoringv1.RelabelConfig) error {
+func validateRelabelConfig(p *monitoringv1.Prometheus, rc monitoringv1.RelabelConfig) error {
 	relabelTarget := regexp.MustCompile(`^(?:(?:[a-zA-Z_]|\$(?:\{\w+\}|\w+))+\w*)+$`)
+	promSpecVersion := semver.MustParse(p.Spec.Version)
+	minimumVersion := promSpecVersion.GTE(semver.MustParse("2.36.0"))
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+
+	if (rc.Action == string(relabel.Lowercase) || rc.Action == string(relabel.Uppercase)) && !minimumVersion {
+		level.Warn(logger).Log("msg", fmt.Sprintf("ignoring %s action not supported by Prometheus version=%s minimum_version=2.36.0", rc.Action, promSpecVersion))
```

the error will be logged by the caller so we should just return the error.